### ISSUE TITLE
CC005 - invalid condition syntax causes error

### ISSUE THREAD

**Issue by valentine195** [apigee/apigeelint#575]
This rule catches thrown errors out of the truth table and it causes CC005 to get applied to invalid condition syntax; should that be the case, or should these get CC007 applied to them?

Here's an example from a real internal proxy:

```xml
<Condition>
    request.verb IsNot "GET"
    and !proxy.pathsuffix MatchesPath "/path"
</Condition>
```

The negation is incorrect, but this was returned as CC005 Unterminated Strings for invalid syntax, which caused confusion for the owning team.

I also noticed while looking into it that the AST returns `undefined` for an empty Condition (like this: `<Condition/>` ), which also causes this rule to fail.

**Comment by DinoChiesa** [comment]
Thanks for reporting this.
